test/harry/main/org/apache/cassandra/harry/dsl/OverridingCkGenerator.java

```diff
@@ -0,0 +1,187 @@
+    private final Map<ArrayWrapper, Long> valueToDescriptor;
+
+    // Had to be a static method because you can not call super after you have initialised any fields
+    public static OverridingCkGenerator make(DataGenerators.KeyGenerator delegate)
+    {
+        KeyPartOverride[] columnValueOverrides = new KeyPartOverride[delegate.columns.size()];
+        List<ColumnSpec<?>> columnSpecOverrides = new ArrayList<>();
+        for (int i = 0; i < delegate.columns.size(); i++)
+        {
+            columnValueOverrides[i] = new KeyPartOverride<>(delegate.columns.get(i).generator());
+            columnSpecOverrides.add(delegate.columns.get(i).override(columnValueOverrides[i]));
+        }
+        assert columnValueOverrides.length == columnSpecOverrides.size();
+        return new OverridingCkGenerator(delegate, columnValueOverrides, columnSpecOverrides);
+    }
+
+    private OverridingCkGenerator(DataGenerators.KeyGenerator delegate,
+                                  KeyPartOverride[] columnValueOverrides,
+                                  List<ColumnSpec<?>> columnSpecOverrides)
+    {
+        super(columnSpecOverrides);
+        this.columnValueOverrides = columnValueOverrides;
+        this.columnSpecOverrides = columnSpecOverrides;
+        this.delegate = delegate;
+        this.valueToDescriptor = new HashMap<>();
+    }
+
+    public void override(long descriptor, Object[] value)
+    {
+        long[] slices = delegate.slice(descriptor);
+        for (int i = 0; i < slices.length; i++)
+            columnValueOverrides[i].override(slices[i], value[i]);
+
+        // We _always_ deflate clustering key as a package, since we can not validate a clustering key without all components anyways.
+        valueToDescriptor.put(new ArrayWrapper(value), descriptor);
+    }
+
+    @Override
+    public Object[] inflate(long descriptor)
+    {
+        return DataGenerators.inflateKey(columnSpecOverrides, descriptor, slice(descriptor));
+    }
+
+    @Override
+    public long deflate(Object[] value)
+    {
+        Long descriptor = valueToDescriptor.get(new ArrayWrapper(value));
+        if (descriptor != null)
+            return descriptor;
+
+        return delegate.deflate(value);
+    }
+
+    @Override
+    public int byteSize()
+    {
+        return delegate.byteSize();
+    }
+
+    @Override
+    public int compare(long l, long r)
+    {
+        return delegate.byteSize();
+    }
+
+    @Override
+    public long[] slice(long descriptor)
+    {
+        return delegate.slice(descriptor);
+    }
+
+    @Override
+    public long stitch(long[] parts)
+    {
+        return delegate.stitch(parts);
+    }
+
+    @Override
+    public long minValue(int idx)
+    {
+        return delegate.minValue(idx);
+    }
+
+    @Override
+    public long maxValue(int idx)
+    {
+        return delegate.maxValue(idx);
+    }
+
+    public static class KeyPartOverride<T> implements Bijections.Bijection<T>
```

Comment:
This shares some code w/ `OverridingBijection`, but I'll let you decide whether there's enough in common to factor something out. Maybe `KeyPartOverride` could be an `OverridingBijection` that just overrides `override()` and `deflate()`...

Reply:
Yeah, good point. Probably the overhead of `valueToDescriptor` is not worth custom class.